src/charm.py

```diff
@@ -99,6 +158,35 @@ def _validated_address(self, address: str) -> str:
 
         return target
 
+    def _is_scheme_valid(self) -> bool:
+        return self._scheme in ("http", "https")
+
+    def _is_tls_ca_valid(self) -> bool:
+        if ca_cert := self._tls_ca_cert:
+            try:
+                # try to convert the cert to another format
+                # an exception wll be raised if ssl fails to do so due to a formatting error
+                ssl.PEM_cert_to_DER_cert(ca_cert)
+                return True
+            except ValueError as e:
+                logger.error("Invalid CA cert provided %s", str(e))
+                return False
+
+        return True
+
+    # EVENT HANDLERS
+    def _on_collect_unit_status(self, event: ops.CollectStatusEvent):
+        """Set unit status depending on the state."""
+        if not self._targets:
+            event.add_status(ops.BlockedStatus("No targets specified, or targets invalid."))
+        if not self._is_scheme_valid():
```

Comment:
for now this is fine, but I'd add a `#TODO: use a pydantic Config object` to do this in a nicer way.